CoreFoundation/PlugIn.subproj/CFBundle_Executable.c

```diff
@@ -69,94 +66,76 @@ static Boolean _binaryLoadable(CFURLRef url) {
+        
         if (executableURL && !_binaryLoadable(executableURL)) {
             CFRelease(executableURL);
             executableURL = NULL;
         }
     }
-#elif DEPLOYMENT_TARGET_WINDOWS
+    
+#if FHS_BUNDLES || FREESTANDING_BUNDLES
     if (!executableURL) {
-        executableURL = CFURLCreateWithFileSystemPathRelativeToBase(kCFAllocatorSystemDefault, exeName, kCFURLWindowsPathStyle, false, urlPath);
-        if (executableURL && !_urlExists(executableURL)) {
+        // Try finding the executable using library prefix/suffix.
+        CFStringRef sharedLibraryName = CFStringCreateWithFormat(kCFAllocatorSystemDefault, NULL, CFSTR("%@%@%@"), _CFBundleSharedLibraryFilenamePrefix, exeName, _CFBundleSharedLibraryFilenameSuffix);
+        
+        executableURL = CFURLCreateWithFileSystemPathRelativeToBase(kCFAllocatorSystemDefault, sharedLibraryName, PLATFORM_PATH_STYLE, false, urlPath);
+        
+        if (!_binaryLoadable(executableURL)) {
             CFRelease(executableURL);
             executableURL = NULL;
         }
+        
+        CFRelease(sharedLibraryName);
     }
+    
     if (!executableURL) {
-        if (!CFStringFindWithOptions(exeName, CFSTR(".dll"), CFRangeMake(0, CFStringGetLength(exeName)), kCFCompareAnchored|kCFCompareBackwards|kCFCompareCaseInsensitive, NULL)) {
-#if defined(DEBUG)
-            CFStringRef extension = CFSTR("_debug.dll");
-#else
-            CFStringRef extension = CFSTR(".dll");
-#endif
-            CFStringRef newExeName = CFStringCreateWithFormat(kCFAllocatorSystemDefault, NULL, CFSTR("%@%@"), exeName, extension);
-            executableURL = CFURLCreateWithString(kCFAllocatorSystemDefault, newExeName, urlPath);
-            if (executableURL && !_binaryLoadable(executableURL)) {
-                CFRelease(executableURL);
-                executableURL = NULL;
-            }
-            CFRelease(newExeName);
-        }
-    }
-    if (!executableURL) {
-        if (!CFStringFindWithOptions(exeName, CFSTR(".exe"), CFRangeMake(0, CFStringGetLength(exeName)), kCFCompareAnchored|kCFCompareBackwards|kCFCompareCaseInsensitive, NULL)) {
-#if defined(DEBUG)
-            CFStringRef extension = CFSTR("_debug.exe");
-#else
-            CFStringRef extension = CFSTR(".exe");
-#endif
-            CFStringRef newExeName = CFStringCreateWithFormat(kCFAllocatorSystemDefault, NULL, CFSTR("%@%@"), exeName, extension);
-            executableURL = CFURLCreateWithString(kCFAllocatorSystemDefault, newExeName, urlPath);
-            if (executableURL && !_binaryLoadable(executableURL)) {
-                CFRelease(executableURL);
-                executableURL = NULL;
-            }
-            CFRelease(newExeName);
+        // Try finding the executable using executable prefix/suffix.
+        CFStringRef executableName = CFStringCreateWithFormat(kCFAllocatorSystemDefault, NULL, CFSTR("%@%@%@"), _CFBundleExecutableFilenamePrefix, exeName, _CFBundleExecutableFilenameSuffix);
```

Comment:
Per the earlier comments: Windows needs `_debug.dll` searching. We should avoid regressions.